src/Reader/Reader.php

```diff
@@ -207,26 +212,27 @@ public static function import($uri, $etag = null, $lastModified = null)
                     $lastModified = $cache->getItem($cacheId . '_lastmodified');
                 }
                 if ($etag) {
-                    $headers->addHeaderLine('If-None-Match', $etag);
+                    $headers['If-None-Match'] = [$etag];
                 }
                 if ($lastModified) {
-                    $headers->addHeaderLine('If-Modified-Since', $lastModified);
+                    $headers['If-Modified-Since'] = [$lastModified];
                 }
             }
-            $response = $client->send();
+            $response = $client->get($uri, $headers);
             if ($response->getStatusCode() !== 200 && $response->getStatusCode() !== 304) {
                 throw new Exception\RuntimeException('Feed failed to load, got response code ' . $response->getStatusCode());
             }
             if ($response->getStatusCode() == 304) {
                 $responseXml = $data;
             } else {
+                $hasHeaders  = $response instanceof Http\HeaderAwareResponseInterface;
```

Comment:
Not necessarily, which is why the check is present. I'll clarify with an updated patch tomorrow.